routemaster/state_machine/tests/test_state_machine_utils.py

```diff
@@ -0,0 +1,105 @@
+import datetime
+
+import mock
+import pytest
+import dateutil
+import freezegun
+
+from routemaster.feeds import Feed
+from routemaster.webhooks import WebhookResult
+from routemaster.state_machine import utils
+from routemaster.state_machine.types import LabelRef
+from routemaster.state_machine.exceptions import UnknownStateMachine
+
+
+def test_get_state_machine(app_config):
+    label = LabelRef(name='foo', state_machine='test_machine')
+    (state_machine,) = app_config.config.state_machines.values()
+    assert utils.get_state_machine(app_config, label) == state_machine
+
+
+def test_get_state_machine_not_found(app_config):
+    label = LabelRef(name='foo', state_machine='nonexistent_machine')
+    with pytest.raises(UnknownStateMachine):
+        utils.get_state_machine(app_config, label)
+
+
+def test_needs_gate_evaluation_for_metadata_change(app_config, create_label):
+    label = create_label('foo', 'test_machine', {})
+    (state_machine,) = app_config.config.state_machines.values()
+
+    with app_config.db.begin() as conn:
+        assert utils.needs_gate_evaluation_for_metadata_change(
+            state_machine,
+            label,
+            {'foo': 'bar'},
+            conn,
+        ) is False
+        assert utils.needs_gate_evaluation_for_metadata_change(
+            state_machine,
+            label,
+            {'should_progress': 'bar'},
+            conn,
+        ) is True
+
+
+def test_needs_gate_evaluation_for_metadata_change_with_action(app_config, create_label, mock_webhook):
```

Comment:
I realise it's part of the common setup of the tests, though I found it non-obvious that this test worked by progressing the user from a gate to an action and forcing the action to fail; thus leaving the user in an action state.

Could we either make that progression more explicitly part of the test setup or assert that the user is in an action state as part of the setup? The latter clarifies the state of the user without needing to worry about how it got there, which might be easier.

Reply:
We could do this, and I was contemplating it, but I think it might be a big refactor of the tests, so I'm keen to leave this for another PR. Thoughts?

Reply:
Would even just an assertion be that complex to add? I think just the assertion alone would clarify the point at which the user is expected to be in the action state. A comment about how we achieve it might be enough to explain the slightly odd create-with-fail behaviour.

I don't think it's worth doing a(nother) big refactor on this PR just for this.

Reply:
ffd5c481fec5b0454c04c5314d57da8bc0e2e95d is that good?